### ISSUE TITLE
accessing self in packages

### ISSUE THREAD

**Issue by olistrik** [snowfallorg/lib#56]
Hi there,

I'm trying to export a package from my flake as both a package, and a namespaced overlay, while also have it easily usable internally. 
Snowfall has made this trivially easy to do by simply adding the derivation to the `packages` folder, however there's a catch, I also need access to a custom set of modules that I would also like to export from my flake.
Exporting it doesn't seem to be an issue, just adding it to the mkFlake closure seems to handle that just fine, however accessing it in packages seems to be impossible unless I manually create them in the outputs-builder. However I then loose the namespaced overlay, and I cannot access the package in my systems.

Unlike snowfall overlays, I don't seem to be able to access `self` in packages, so I'm struggling to work out if I can access the modules at all.

Basically I have this:

flake.nix
```
{
  inputs = ...;
  outputs = inputs: let
    lib = ...;
  in lib.mkFlake {
    overlays = [ inputs.nixvim.default ];
    nixvimModules = lib.snowfall.module.create-modules {
      src = "${./modules/nixvim}";
    }; 
  };
}
```

packages/nixvim/default.nix:

```
{lib, pkgs, nixvim, nixvimModules ? {}}: # I cannot access nixvimModules here.
with lib;
nixvim.makeNixvimWithModule {
  inherit pkgs;
  imports = attrValues nixvimModules;
  
  config = ...;
}
```

**Comment by jakehamilton** [comment]
I think that, ideally, `self` would be supplied along with the other inputs. It was previously removed due to very common infinite recursion, but I will try adding it back for use in packages (and possibly overlays). Can't say for sure that it won't be a footgun, but it's worth a try!
